util/test/paratest.chapcs

```diff
@@ -133,17 +134,27 @@ def get_good_nodepara():
     return nodepara
 
 
-def run_paratest(args):
+def run_paratest(paratest_args, nightly=False):
     """
     Run paratest inside an salloc using all nodes that are not reserved
     exclusively on the chapel partition. Throw `--oversubscribe --nice` and
     turn off affinity and limit how many executables can run at once so we play
     nice with other testing going on.
     """
-    nodepara = get_good_nodepara()
-    num_free_nodes = get_num_non_exclusive_nodes()
-    para_env = ['-env', 'CHPL_TEST_LIMIT_RUNNING_EXECUTABLES=yes', '-env',
-                'CHPL_RT_OVERSUBSCRIBED=yes']
+    if not debug:
+        nodepara = get_good_nodepara()
+        num_free_nodes = get_num_non_exclusive_nodes()
+    else:
+        print('Running in debug mode')
+        print('paratest_args: {}'.format(paratest_args))
+        print('nightly: {}'.format(nightly))
+        nodepara = 1
+        num_free_nodes = 1
+
+    para_env = ['-env', 'CHPL_RT_OVERSUBSCRIBED=yes']
+    # we run exclusively with nightly, so no need to limit
+    if not nightly:
```

Comment:
I think the running executable limiting is more about valgrind running than nightly being used. If we used paratest.chapcs for say nightly gasnet testing, I think we'd still want to limit the running executables. I'd probably base this off `-valgrind[exe]` being in the args -- I think that also makes sense for users doing a paratest.chapcs.

Reply:
Similar to my comment above, as we'll also switch to exclusive allocations for nightly runs, I thought we wouldn't want to limit as there won't be any other job running on the same node to mess with.

But I can still make that change if you want?

Reply:
Even for a single job, we usually want to limit to only running a single executable at a time so that multiple instances of the runtime don't interfere with each other (i.e. if you run 2 `./hello`s there can be significant interference from our runtime, though we limit that with `CHPL_RT_OVERSUBSCRIBED=yes` above)  I think valgrind is the exception to that because valgrind serializes so it can't really cause interference with other executions.